go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -338,6 +339,32 @@ func (bsnc *BaseSecondaryNetworkController) addLogicalPortToNetworkForNAD(pod *k
 		ops = append(ops, addOps...)
 	}
 
+	if util.IsNetworkSegmentationSupportEnabled() && bsnc.IsPrimaryNetwork() && config.Gateway.DisableSNATMultipleGWs {
```

Comment:
shouldn't this be part of the gateway code ?

Reply:
nope gateway code doesn't have podIPs.. this has to be part of pod code unfortunately.

Reply:
but can't we move this func there and wrap it in a method ? It would do exactly this, and receive the pod IPs from the controller. 

not a must.

Reply:
pod LSP add + snat add should be in same transact...

Reply:
> pod LSP add + snat add should be in same transact.

Have the method return the ops.

I mostly would rather not have all these details shoved in the code's reader face whenever they read the add logical port flow.

Reply:
yea please note this is not the final version see my PR message "need to clean up some things " .. for sure this will a util.. I will see if it belongs in gatewayconfig or not..but this is not the final version of this PR.

Reply:
hmm I am trying to move this to gatewayconfig let's take L2 or L3 for example, I think you meant:
```
type SecondaryL2GatewayConfig struct {
	config      *util.L3GatewayConfig
	hostSubnets []*net.IPNet
	gwLRPIPs    []*net.IPNet
	externalIPs []net.IP
}

func (oc *SecondaryLayer2NetworkController) nodeGatewayConfig(node *corev1.Node) (*SecondaryL2GatewayConfig, error) {
	l3GatewayConfig, err := util.ParseNodeL3GatewayAnnotation(node)
```
to add it as a util to this config and also L3 right? let's talk monday in our sync to see if I can figure this out with you

but I have pushed a new version where I have cleaned up the addLSP bits and broken the logic into smaller piece dunno if that is better or not, but happy to hear your thoughts..

Reply:
the code similar to addLogicalPort code in default controller: https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/ovn/pods.go#L299

It seems OK to me.

Reply:
yeah will talk to @maiqueb on monday to see what he meant.